Userland/Libraries/LibC/string.cpp

```diff
@@ -368,6 +368,23 @@ char* strstr(const char* haystack, const char* needle)
     return const_cast<char*>(haystack);
 }
 
+char* strcasestr(const char* haystack, const char* needle)
+{
+    char nch;
+    char hch;
+
+    if ((nch = *needle++) != 0) {
+        size_t len = strlen(needle);
+        do {
+            do {
+                if ((hch = *haystack++) == 0)
+                    return nullptr;
+            } while (hch != nch);
```

Comment:
Won't this fast-forward beyond a potential starting point of the needle, as it is case-sensitive?

Reply:
I don't think so, the `case` class of functions are not case sensitive while the standard ones are.

Reply:
Sure, the functions are not, but raw comparisons of `char` are. More precisely, you are missing `tolower` or something similar around those character comparisons here.